lib/chef/provider/group/groupadd.rb

```diff
@@ -126,7 +126,8 @@ def set_options
 
         def groupadd_options
           opts = ""
-          opts << " -r" if @new_resource.system
+          # Solaris doesn't support system groups.
+          opts << " -r" if @new_resource.system && !node.platform?("solaris2")
```

Comment:
Think I should add `, "omnios"` here? I don't know those Solarish platforms well enough.